src/utils/parsers.rs

```diff
@@ -20,9 +22,43 @@ pub fn parse_schema_source(loc: &str) -> Result<SchemaSource> {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct GraphRef {
+    pub name: String,
+    pub variant: String,
+}
+
+/// NOTE: THIS IS A TEMPORARY SOLUTION. IN THE FUTURE, ALL GRAPH ID PARSING
+/// WILL HAPPEN IN THE BACKEND TO KEEP EVERYTHING CONSISTENT. THIS IS AN
+/// INCOMPLETE PLACEHOLDER, AND MAY NOT COVER EVERY SINGLE VALID USE CASE
+pub fn parse_graph_ref(graph_id: &str) -> Result<GraphRef> {
+    let pattern = Regex::new(r"^[a-zA-Z][a-zA-Z0-9_-]{0,63}$").unwrap();
+    let variant_pattern = Regex::new(r"^([a-zA-Z][a-zA-Z0-9_-]{0,63})@(.{0,63})$").unwrap();
+
+    let valid_graph_name_only = pattern.is_match(graph_id);
+    let valid_graph_with_variant = variant_pattern.is_match(graph_id);
+
+    if valid_graph_name_only {
+        Ok(GraphRef {
+            name: graph_id.to_string(),
+            variant: "current".to_string(),
+        })
+    } else if valid_graph_with_variant {
+        let matches = variant_pattern.captures(graph_id).unwrap();
+        let name = matches.get(1).unwrap().as_str();
+        let variant = matches.get(2).unwrap().as_str();
+        Ok(GraphRef {
+            name: name.to_string(),
+            variant: variant.to_string(),
+        })
+    } else {
+        Err(anyhow!("Graph IDs must be in the format <NAME> or <NAME>@<VARIANT>, where <NAME> can only contain letters, numbers, or the characters `-` or `_`, and must be 64 characters or less. <VARIANT> must be 64 characters or less."))
```

Comment:
Both `pattern` and `variant_pattern` seem to be checking for both validity of the string, character count, and the types of characters used.

To give user much more detailed instructions, I am wondering if we should independently check for character count in `variant` and `name` and error earlier with the exact information that's missing from `graph_id`. It would make this parser a bit lengthier than you have it now, but we will be able to distinguish between:
- Graph IDs must be in format `<NAME>` or `<NAME>@<VARIANT>`, but {x} was provided.
- `<NAME>` can only contain these certain characters, but {x} was provided.
- `<NAME>` can only be 64 characters long, but {x number} was provided.
- `<VARIANT>` can only be 64 characters long, but {x number} was provided.